test/framework/http_utils_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework_test
+
+import (
+	"context"
+	"github.com/gardener/gardener/test/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"net/http"
+	"net/http/httptest"
+)
+
+var _ = Describe("HTTP Utils tests", func() {
+
+	It("Should perform a basic http get request", func() {
+		var called int
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
It's better to create a test scoped variable 

```go
handlerFunc func(w http.ResponseWriter, r *http.Request)
```

And have 

```go
JustBeforeEach(func(){
   // server is test-scoped variable
   server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    handlerFunc(w,r)
  })
})

AfterEach(func(){
  server.Close()
})
```

Then the test can be simplified to 

```go
It("Should perform a basic http get request", func() {
    var called int
    handlerFunc = func(w http.ResponseWriter, r *http.Request) {
        Expect(r.Method).To(Equal(http.MethodGet))
        called++
        _, err := w.Write(nil)
        Expect(err).ToNot(HaveOccurred())
    })
    
    _, err := framework.HTTPGet(context.TODO(), server.URL)
    Expect(err).ToNot(HaveOccurred())

    Expect(called).To(Equal(1))
})
```